app/src/main/java/org/mozilla/fenix/experiments/NimbusSetup.kt

```diff
@@ -63,7 +63,10 @@ fun createNimbus(context: Context, url: String?): NimbusApi {
             // Note: Using BuildConfig.BUILD_TYPE is important here so that it matches the value
             // passed into Glean. `Config.channel.toString()` turned out to be non-deterministic
             // and would mostly produce the value `Beta` and rarely would produce `beta`.
-            channel = BuildConfig.BUILD_TYPE
+            channel = BuildConfig.BUILD_TYPE,
+            customTargetingAttributes = mapOf(
+                "isFirstRun" to context.settings().isFirstRun.toString()
```

Comment:
@mcomella this looks like it will affect our startup time, but I see above that we touch shared prefs multiple times as part of Nimbus initialization already. Not sure if you're already aware of this since it's required for first run usage with Nimbus.

Reply:
This is very good to know, if there is a less impactful way of storing and retrieving this I'd be happy to rework this to be as minimally impactful to startup as possible.

Reply:
@travis79 I think I know the answer to this already, but to confirm, the "first run" is needed for visual features that are needed on startup, is that right?

Reply:
Right, the idea behind this is the ability to target first run users for experiments.

Reply:
Thanks for the ping – we can benchmark the difference in start up time. I added something to our board for MarcLeclair and I to triage. Is there anything special I need to get this code to run? e.g. add an API key or something?

Reply:
@mcomella I don't think you need to do anything special to get this to run, it should run as part of normal init of Nimbus. Regardless of whether it is the actual first-run of the app or not, it should at least read the pref every launch. On the very first run, it would also write to the pref, so there may be some slight additional overhead difference between the very first run, and subsequent runs. Wiping the app data, or performing a fresh install, should allow you to trigger the "first-run" behavior which includes writing to the pref.

Reply:
@mcomella @jonalmeida It looks like we are shifting priorities a bit on the first-run targeting, so less pressing to get this resolved soon, but I would appreciate an update on the timeframe for this (or even if there is something I could profile or do myself to help unblock this).
 
Thanks! 
 
cc @jhugman